aries_cloudagent/protocols/present_proof/v2_0/formats/dif/handler.py

```diff
@@ -151,9 +151,16 @@ async def create_bound_request(
             A tuple (updated presentation exchange record, presentation request message)
 
         """
-        dif_proof_request = pres_ex_record.pres_proposal.attachment(
+        dif_proof_request = {}
+        pres_proposal_dict = pres_ex_record.pres_proposal.attachment(
             DIFPresFormatHandler.format
         )
+        if "options" not in pres_proposal_dict:
+            dif_proof_request["options"] = {"challenge": str(uuid4())}
+        else:
+            dif_proof_request["options"] = pres_proposal_dict["options"]
```

Comment:
Should we add a challenge value if there isn't one?

Reply:
Without a `challenge` we cannot verify the presentation.

Reply:
Right, but at the moment we seem to generate a challenge value only if `options` isn't provided?

Reply:
Resolved [here](https://github.com/hyperledger/aries-cloudagent-python/blob/b3fe34fc75fbfb3c3896464bc5b7d832c231487c/aries_cloudagent/protocols/present_proof/v2_0/formats/dif/handler.py#L163)